### PR TITLE
D8NID-290: Theme settings for social links.

### PR DESCRIPTION
**Pull request by johangant**
- Add settings for defined social media platform links.
- Surface to theme as render array elements in menu and footer region.

### CODE REVIEW

File: nicsdru_nidirect_theme.theme [hunk 1]
```diff
@@ -6,7 +6,90 @@
...
+ */
+function nicsdru_nidirect_theme_form_system_theme_settings_alter(&$form, FormStateInterface &$form_state, $form_id = NULL) {
+  // Work-around for a core bug affecting admin themes. See issue #943212.
+  if (isset($form_id)) {
+    return;
+  }
+
+  // Add a theme settings group to gather social media link URLs for
+  // presentation in the site header/footer regions.
+  $form['nidirect_sst'] = [
```
Comment (omahm) [inline nicsdru_nidirect_theme.theme]: what is 'sst'? I'd avoid abbreviations if possible.

File: nicsdru_nidirect_theme.theme [hunk 2]
```diff
@@ -6,7 +6,90 @@
...
+/**
+ * Implements hook_preprocess_page() for page.html.twig.
+ */
+function nicsdru_nidirect_theme_preprocess_page(array &$variables) {
+  // Links for: Twitter, Facebook, Youtube and RSS.
+  $links['twitter'] = Link::fromTextAndUrl(t('Twitter'),
+    Url::fromUri(theme_get_setting('nidirect_sst_twitter'), [
+      'attributes' => ['title' => t('Twitter')]
+    ]))->toRenderable();
+  $links['fb'] = Link::fromTextAndUrl(t('Facebook'),
```
Comment (omahm) [inline nicsdru_nidirect_theme.theme]: I'd switch 'fb' to 'facebook' to keep it consistent with the others.

File: nicsdru_nidirect_theme.theme [hunk 3]
```diff
@@ -6,7 +6,90 @@
...
 use Drupal\user\Entity\User;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_preprocess_page() for page.html.twig.
+ */
+function nicsdru_nidirect_theme_preprocess_page(array &$variables) {
+  // Links for: Twitter, Facebook, Youtube and RSS.
+  $links['twitter'] = Link::fromTextAndUrl(t('Twitter'),
+    Url::fromUri(theme_get_setting('nidirect_sst_twitter'), [
```
Comment (omahm) [inline nicsdru_nidirect_theme.theme]: What is theme_get_setting  returns NULL because nidirect_sst_twitter isn't set?

**Comment by David-Corbett** [comment]
I think that all this functionality should be added to the nicsdru_origins_theme and not this repo as this is shared functionality across nics sites and would be invoked any site that is sub themed from the origins theme

**Comment by johangant** [comment]
@David-Corbett yep, that's not a problem to move it.

@omahm let me expand those variable names and handle the nulls, will submit those changes in the new PR.

Thanks, both.

**Pull request closed without merging**